wandb_logger.py

```diff
@@ -0,0 +1,58 @@
+"""
+A simple module to initialize wandb and define log functionalities for catlearn
+ models
+"""
+from typing import Any
+from collections import defaultdict
+from itertools import chain
+import wandb
+
+from catlearn.composition_graph import NodeType, ArrowType, DirectedGraph
+from catlearn.relation_cache import RelationCache
+from catlearn.categorical_model import TrainableDecisionCatModel
+
+project = 'catlearn'
+config = {}
+
+
+wandb.login()
+wandb.init(project=project, config=config)
+
+def log_results(
+    cache: RelationCache[NodeType, ArrowType],
+    matched: DirectedGraph[NodeType],
+    **info_to_log: Any) -> None:
+    """
+        Log results from a training step
+    """
+
+    # costs
+    total = sum(
+        sum(elem for _, elem in costs.values())
+        for costs in matched.edges.values())
+    nb_labels = sum(len(costs) for costs in matched.edges.values())
+    cost_per_label = total/max(nb_labels, 1) + cache.causality_cost
+
+    # arrow by order and causality check
+    used_arrows = set(chain(*((
+        arr.suspend(src, tar) for (arr, _) in content.values())
+        for src, tar, content in matched.edges(data=True))))
+    causal_arrows = set(cache.arrows(include_non_causal=False))
+    noncausal_arrows = set(cache.arrows()) - causal_arrows
+
+    arrows = defaultdict(lambda: {'used': 0, 'causal': 0, 'noncausal': 0})
+    for arr in used_arrows:
+        arrows[len(arr)]['used'] += 1
+    for arr in causal_arrows:
+        arrows[len(arr)]['causal'] += 1
+    for arr in noncausal_arrows:
+        arrows[len(arr)]['noncausal'] += 1
+
+    wandb.log({
+        'nb_labels': nb_labels,
+        'total_match_cost': total,
+        'total_causality_cost': cache.causality_cost,
+        'cost_per_label': cost_per_label,
+        'arrow_numbers': arrows,
+        **info_to_log
```

Comment:
Perhaps adding a check we won't override keys here? Would prevent stupid mistakes :D